hive.go

```diff
@@ -69,10 +71,14 @@ func main() {
 		simList = nil
 	}
 
+	dockerRegistries := strings.Split(strings.Trim(*dockerRegistriesRaw, " "), ",")
+
 	// Create the docker backends.
 	dockerConfig := &libdocker.Config{
-		Inventory:   inv,
-		PullEnabled: *dockerPull,
+		Inventory:        inv,
+		PullEnabled:      *dockerPull,
+		DockerRegistries: dockerRegistries,
+		AuthType:         libdocker.AuthType(*dockerAuthType),
```

Comment:
Introducing the `AuthType` ends up creating a lot of boilerplate code. What you want to achieve is using the `cred-helper` if it is defined.

Let's just simplify this whole setting into a boolean `UseCredHelper` (and flag `--docker.cred-helper`) which will configure the authentication for all registries when set.